src/Microsoft.SqlTools.ServiceLayer/AzureBlob/BlobService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Microsoft.SqlServer.Management.Common;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.Hosting.Protocol;
+using Microsoft.SqlTools.ServiceLayer.AzureBlob.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+
+namespace Microsoft.SqlTools.ServiceLayer.AzureBlob
+{
+    class BlobService
+    {
+        private static readonly Lazy<BlobService> instance = new Lazy<BlobService>(() => new BlobService());
+        private static ConnectionService connectionService = null;
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
+        {
+            get
+            {
+                if (connectionService == null)
+                {
+                    connectionService = ConnectionService.Instance;
+                }
+                return connectionService;
+            }
+            set
+            {
+                connectionService = value;
+            }
+        }
+
+        /// <summary>
+        /// Default, parameterless constructor.
+        /// </summary>
+        internal BlobService()
+        {
+        }
+
+        /// <summary>
+        /// Gets the singleton instance object
+        /// </summary>
+        public static BlobService Instance
+        {
+            get { return instance.Value; }
+        }
+
+        public void InitializeService(IProtocolEndpoint serviceHost)
+        {
+
+            // Create shared access secret
+            serviceHost.SetRequestHandler(CreateSasRequest.Type, HandleCreateSasRequest);
+        }
+
+        internal async Task HandleCreateSasRequest(
+           CreateSasParams optionsParams,
+           RequestContext<CreateSasResponse> requestContext)
+        {
+            try
+            {
+                ConnectionInfo connInfo;
+                bool supported = IsBackupRestoreOperationSupported(optionsParams.OwnerUri, out connInfo);
+                var response = new CreateSasResponse();
+
+                if (supported && connInfo != null)
+                {
+                    SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "Backup");
```

Comment:
please put the disposable objects you are creating here into using statements.

Reply:
I agree about sqlConn. However, I'm not sure serverConnection and sqlServer are disposable objects.